### ISSUE TITLE
Update to latest client

### ISSUE THREAD

**Issue by k4n30** [flathub/com.amazon.Workspaces#49]
Hi, is it possible to update to the new WorkSpaces client (2023) - https://d3nt0h4h6pmmc4.cloudfront.net/new_workspacesclient_jammy_amd64.deb as per https://clients.amazonworkspaces.com/linux-install

I understand it may deprecate PCoIP WorkSpaces, but I'm pretty sure I read somewhere AWS weren't putting any development effort into them anymore anyway with the new focus on the WSP protocol

**Comment by bbhtt** [comment]
I'll add it to the `flathub-beta` repository if successful. The stable will continue to have 'classic` workspace. You can enable beta repository using https://discourse.flathub.org/t/how-to-use-flathub-beta/2111 and install it. 

Please test, the test build in https://github.com/flathub/com.amazon.Workspaces/pull/50 once done and let me know if it works. Thanks!

**Comment by bbhtt** [comment]
I'm wondering whether the .deb from above works for you? It seems to crash immediately on my system, and so does the above flatpak build.

**Comment by bbhtt** [comment]
Ok it works for me in a Ubuntu 22.04 VM. It seems the postinst script inside the deb have a bunch of hacks that need to be integrated in the flatpak to make it run.

This being extra-data makes the job harder... 

I'll try to make it work but no promises.

**Comment by bbhtt** [comment]
Ok, fixed that issue by this hack `export GDK_PIXBUF_MODULE_FILE=/app/workspacesclient/usr/lib/x86_64-linux-gnu/workspacesclient/dcv/gdk-pixbuf-2.0/2.10.0/loaders.cache`


The deb has this:

```
export LD_LIBRARY_PATH="/app/workspacesclient/usr/lib/x86_64-linux-gnu/workspacesclient/dcv:${LD_LIBRARY_PATH}"
```
which I'm also doing

But it can't find libraries the in `/usr/lib` (from the runtime) at all. Thus either crashes or has no network.

Also, for some weird reason it ships with `libsoup-3.0.so.0` inside the deb yet depends on webkit2gtk-4.0` which links against `libsoup-2.4` and the two libsoup versions are incompatible.

**Comment by bbhtt** [comment]
https://github.com/flathub/com.amazon.Workspaces/pull/52 is a "working" build but network doesn't work, I'm not sure why. I suspect it could be due to the libsoup2.4/libsoup3 mixing.

Help appreciated in figuring that out. Logs don't show much other than "install glib-networking" but that is already in the runtime.

**Comment by compuguy** [comment]
@k4n30 There are still plenty of end users using PCoIP so I wouldn't use that version. I think the best path forward is to use the 20.04 client for now @bbhtt if it doesn't have the issues with `glib-networking` like #52 had.
Source deb: https://d3nt0h4h6pmmc4.cloudfront.net/new_workspacesclient_focal_amd64.deb

**Comment by dylanmtaylor** [comment]
The current client version is very dated and does not support WSP workspaces which means that Ubuntu VDIs are not streamable at all. As Amazon is moving all new Windows and Ubuntu workspaces to WSP, we need to go to the new version for this flatpak to be usable at all.

**Comment by compuguy** [comment]
> The current client version is very dated and does not support WSP workspaces which means that Ubuntu VDIs are not streamable at all. As Amazon is moving all new Windows and Ubuntu workspaces to WSP, we need to go to the new version for this flatpak to be usable at all.

Agreed. On the flip side, the client I work with hasn't transitioned over to WSP workstations and is still using PCOIP. That puts me in a bizarre rock in a hard place situation. I have a personal workspace that's WSP, but the client I work with uses PCOIP...

**Comment by anibyl** [comment]
When trying to install the beta, it fails:

```sh
$ flatpak install flathub-beta com.amazon.Workspaces
...
error: Failed to install com.amazon.Workspaces: While downloading https://d3nt0h4h6pmmc4.cloudfront.net/ubuntu/dists/focal/main/binary-amd64/workspacesclient_4.6.0.4187_amd64.deb: Server returned status 403
```

Unfortunately, I cannot use the stable version anymore, perhaps because my workspace was reinstalled and it uses a new protocol.

**Comment by bbhtt** [comment]
The beta branch is a leftover, and amazon deletes old debs so it is expected that it can't be installed.

The new version does not work https://github.com/flathub/com.amazon.Workspaces/pull/52#issuecomment-2173549057, anyone is welcome to figure out/fix the issue.

If the amazon deletes the deb of the current version and we can't update to the new non pcoip version the package will be retired.

I can't reupload or archive a version: https://clients.amazonworkspaces.com/app-terms.html

> You may not, and you will not encourage, assist or authorize any other person to, (b)... *distribute*, ... the Application

This Flatpak package specifically uses `extra-data` so that the deb gets downloaded on each user's machine and avoid redistribution. Reuploading/archiving a copy would break that clause.

**Comment by compuguy** [comment]
> The beta branch is a leftover, and amazon deletes old debs so it is expected that it can't be installed.
> 
> The new version does not work https://github.com/flathub/com.amazon.Workspaces/pull/52#issuecomment-2173549057, anyone is welcome to figure out/fix the issue.
> 
> If the amazon deletes the deb of the current version and we can't update to the new non pcoip version the package will be retired.
> 
> I can't reupload or archive a version: https://clients.amazonworkspaces.com/app-terms.html
> 
> > You may not, and you will not encourage, assist or authorize any other person to, (b)... *distribute*, ... the Application
> 
> This Flatpak package specifically uses `extra-data` so that the deb gets downloaded on each user's machine and avoid redistribution. Reuploading/archiving a copy would break that clause. 

Yeah I don't get that one clause. I'm pretty sure there are (unnamed) companies and government agencies who distribute it internally.

Anyway, the last time I tried getting either build working in a flatpak in #63 and #62 it just ended in frustration. 😩

**Comment by anibyl** [comment]
Amazon WorkSpaces has a web app: https://webclient.amazonworkspaces.com/

_Maybe this will help people who are unable to use the new client, e.g. those who use an RPM-based distro._

**Comment by compuguy** [comment]
> Amazon WorkSpaces has a web app: https://webclient.amazonworkspaces.com/
> 
> _Maybe this will help people who are unable to use the new client, e.g. those who use an RPM-based distro._

As long as the workspaces tenant has the webclient enabled, yes it is an option. Unfortunately the tenants I use do not have this option enabled, so it doesn't work in my circumstance.

**Comment by ctubbsii** [comment]
The 4.7 version is now reporting that it is out of date and will reach EOL on March 31, 2025, so it's kind of important to have an update available.

**Comment by jianpan** [comment]
Archlinux has ported the latest version of the aws workspace client and even patched back in PCoIP support. Can we wrap this version in flatpak?

https://aur.archlinux.org/packages/amazon-workspaces-bin

**Comment by Clifra-Jones** [comment]
You will have to go with the web client, which means you have to have your workspace on DCV(WPS) and you administrator must enable the web client.
The downside for me is it is painfully slow to connect and sometimes unreliable (doesn't always connect) over VPN.
Why amazon does not support RPM is beyond me!

**Comment by anibyl** [comment]
Even their DEB support is pretty bad. The latest supported distro is Ubuntu 22.04, which is not even the current LTS version.

**Comment by jianpan** [comment]
> > Archlinux has ported the latest version of the aws workspace client and even patched back in PCoIP support. Can we wrap this version in flatpak?
> > https://aur.archlinux.org/packages/amazon-workspaces-bin
> 
> Interesting...so that's how they check to see if PCoIP support works. It might be worth a shot. I don't know.

I installed it in a manjaro box. PCoIP indeed works with this version.

![Image](https://github.com/user-attachments/assets/eb53986b-a44a-49af-b4ab-443cda3de427)
